packages/permission-log-controller/package.json

```diff
@@ -0,0 +1,57 @@
+{
+  "name": "@metamask/permission-log-controller",
+  "version": "5.0.2",
+  "description": "Controller with middleware for logging requests and responses to restricted and permissions-related methods",
+  "keywords": [
+    "MetaMask",
+    "Ethereum"
+  ],
+  "homepage": "https://github.com/MetaMask/core/tree/main/packages/permission-log-controller#readme",
+  "bugs": {
+    "url": "https://github.com/MetaMask/core/issues"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/MetaMask/core.git"
+  },
+  "license": "MIT",
```

Comment:
If we change the license format, we also have to change this. In this case [we have to use a special string](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#license) since we have a custom license:

``` suggestion
  "license": "SEE LICENSE IN LICENSE",
```

Note that this will cause `yarn constraints` to fail, because it's checking that for all packages, the `license` field here is "MIT". To get around this we need to update the constraints.

Merge the latest changes from `main` into this branch. Now in `constraints.pro`, look for this section:

``` prolog
% The license for all published packages must be MIT unless otherwise specified.
gen_enforced_field(WorkspaceCwd, 'license', 'MIT') :-
  \+ workspace_field(WorkspaceCwd, 'private', true),
  WorkspaceCwd \= 'packages/json-rpc-engine',
  WorkspaceCwd \= 'packages/eth-json-rpc-provider'.
% The following published packages use an ISC license instead of MIT.
gen_enforced_field(WorkspaceCwd, 'license', 'ISC') :-
  \+ workspace_field(WorkspaceCwd, 'private', true),
  (
    WorkspaceCwd == 'packages/json-rpc-engine' ;
    WorkspaceCwd == 'packages/eth-json-rpc-provider'
  ).
% Non-published packages do not have a license.
gen_enforced_field(WorkspaceCwd, 'license', null) :-
  workspace_field(WorkspaceCwd, 'private', true).
```

and change it to:

``` prolog
% The license for all published packages must be MIT unless otherwise specified.
gen_enforced_field(WorkspaceCwd, 'license', 'MIT') :-
  \+ workspace_field(WorkspaceCwd, 'private', true),
  WorkspaceCwd \= 'packages/json-rpc-engine',
  WorkspaceCwd \= 'packages/eth-json-rpc-provider',
  WorkspaceCwd \= 'packages/permission-log-controller'.
% The following published packages use an ISC license instead of MIT.
gen_enforced_field(WorkspaceCwd, 'license', 'ISC') :-
  \+ workspace_field(WorkspaceCwd, 'private', true),
  (
    WorkspaceCwd == 'packages/json-rpc-engine' ;
    WorkspaceCwd == 'packages/eth-json-rpc-provider' ;
    WorkspaceCwd == 'packages/permission-log-controller'
  ).
% The following published packages use a custom license instead of MIT.
gen_enforced_field(WorkspaceCwd, 'license', 'SEE LICENSE IN LICENSE') :-
  \+ workspace_field(WorkspaceCwd, 'private', true),
  WorkspaceCwd == 'packages/permission-log-controller'.
% Non-published packages do not have a license.
gen_enforced_field(WorkspaceCwd, 'license', null) :-
  workspace_field(WorkspaceCwd, 'private', true).
```

Reply:
I think the `permission-log-controller` entry under `ISC` needs to be removed for the above to work:

```prolog
% The license for all published packages must be MIT unless otherwise specified.
gen_enforced_field(WorkspaceCwd, 'license', 'MIT') :-
  \+ workspace_field(WorkspaceCwd, 'private', true),
  WorkspaceCwd \= 'packages/json-rpc-engine',
  WorkspaceCwd \= 'packages/eth-json-rpc-provider',
  WorkspaceCwd \= 'packages/permission-log-controller'.
% The following published packages use an ISC license instead of MIT.
gen_enforced_field(WorkspaceCwd, 'license', 'ISC') :-
  \+ workspace_field(WorkspaceCwd, 'private', true),
  (
    WorkspaceCwd == 'packages/json-rpc-engine' ;
    WorkspaceCwd == 'packages/eth-json-rpc-provider'
  ).
% The following published packages use a custom license instead of MIT.
gen_enforced_field(WorkspaceCwd, 'license', 'SEE LICENSE IN LICENSE') :-
  \+ workspace_field(WorkspaceCwd, 'private', true),
  WorkspaceCwd == 'packages/permission-log-controller'.
% Non-published packages do not have a license.
gen_enforced_field(WorkspaceCwd, 'license', null) :-
  workspace_field(WorkspaceCwd, 'private', true).
```

Reply:
Er, sorry, yes, that's right.